src/common/resourceutil/roundtriptime.go

```diff
@@ -61,8 +61,13 @@ func processRTT() {
 				}
 				go func(info netDB.NetworkInfo) {
 					result := selectMinRTT(ch, totalCount)
-					info.RTT = result
-					netDBExecutor.Update(info)
+					if info.RTT < 0 && result < 0 {
```

Comment:
@suresh-lc PTAL. It seems to me that it prevents undesired iteration of discovery trials for the dead devices, so that it improves the efficiency of energy consumption and data storage of the edge device. Current proposal is based on the fixed policy (like allowing 1 iteration once it fails the discovery) and in the mid-long term approach, we might need to develop it based on sth like AI/ML. So my opinion is okay for accepting this as the 1st proposal.

Reply:
Checking on the issue. But i do find this in discovery.go (line 364)
if data.TTL == 0 {
					deleteDevice(data.DeviceID)
					continue
				}
So it deletes a device whose TTL is 0 while multicasting.

Reply:
@suresh-lc Thank you for the comment. 
I think that source code deletes a device that respond to multicast with TTL=0.
However this issue is not related on the code since offline devices couldn't respond to the multicast and the deviceDetectionRoutine function does not check netDB to delete them.

Reply:
Consider a scenario of a device becomes offline for short while and again comes alive. Going with the suggested approach, we would be deleting the device from netDB and again add. And if this happens frequently, then unnecessary delete and add happens. And that too if its too regular, its overhead. So might be we can add a timers and re-try count of 2 or so. What is your opinion.

Reply:
@suresh-lc It makes sense. 
RTT ping cycle is 5 sec. How about giving it 12 chances (around 1min)? It might be good to have a limit variable on the code as well.

Reply:
Guess 1min approach is good. Just to verify completely, you can check on below :
Scenario:
1. Device 1 and device 2 are there .

2. Both discovered each other.

3. Now device 2 is disconnected from network so, device 1 is not able to ping it. ( as per  his logic he will remove device 2 from discovered device list of device 1) 

3. Device 2 again connected to network...is device 1 able to discover it freshly?

If this we tested, we are good to go ahead and check in the changes.